src/components/auction/OngoingAuctionCard.tsx

```diff
@@ -0,0 +1,403 @@
+  PositionsEstimate,
+  RequestType,
+  SubmitArgs,
+} from '@blend-capital/blend-sdk';
+import { Box, Typography, useTheme } from '@mui/material';
+import { Asset, SorobanRpc } from '@stellar/stellar-sdk';
+import Image from 'next/image';
+import { useMemo, useState } from 'react';
+import { useWallet } from '../../contexts/wallet';
+import {
+  useBackstop,
+  useHorizonAccount,
+  usePoolOracle,
+  usePoolUser,
+  useTokenBalance,
+} from '../../hooks/api';
+import { RPC_DEBOUNCE_DELAY, useDebouncedState } from '../../hooks/debounce';
+import { calculateAuctionOracleProfit } from '../../utils/auction';
+import { toBalance, toCompactAddress } from '../../utils/formatter';
+import { requiresTrustline } from '../../utils/horizon';
+import { getErrorFromSim, SubmitError } from '../../utils/txSim';
+import { AnvilAlert } from '../common/AnvilAlert';
+import { DividerSection } from '../common/DividerSection';
+import { OpaqueButton } from '../common/OpaqueButton';
+import { PoolComponentProps } from '../common/PoolComponentProps';
+import { Row } from '../common/Row';
+import { Section, SectionSize } from '../common/Section';
+import { StackedTextBox } from '../common/StackedTextBox';
+import { TxOverview } from '../common/TxOverview';
+import { Value } from '../common/Value';
+import { ValueChange } from '../common/ValueChange';
+import { BidList } from './BidList';
+import { LotList } from './LotList';
+
+export interface OngoingAuctionCardProps extends PoolComponentProps {
+  pool: Pool;
+  auction: Auction;
+  currLedger: number;
+  index: number;
+}
+
+export const OngoingAuctionCard: React.FC<OngoingAuctionCardProps> = ({
+  pool,
+  auction,
+  sx,
+  currLedger,
+}) => {
+  const theme = useTheme();
+  const { walletAddress, connected, poolSubmit, createTrustlines, isLoading, txType } = useWallet();
+  const { data: poolOracle } = usePoolOracle(pool);
+  const { data: backstop } = useBackstop();
+  const { data: horizonAccount } = useHorizonAccount();
+  const { data: poolUser } = usePoolUser(pool);
+  const [simResponse, setSimResponse] = useState<SorobanRpc.Api.SimulateTransactionResponse>();
+  const [parsedSimResult, setParsedSimResult] = useState<Positions>();
+  const [loadingEstimate, setLoadingEstimate] = useState<boolean>(false);
+
+  const scaledAuction = auction.scale(currLedger)[0];
+  const assetKeys = useMemo(() => {
+    if (auction.type === AuctionType.Interest) {
+      const assets = new Set(scaledAuction.data.lot.keys());
+      for (const asset of Array.from(scaledAuction.data.bid.keys())) {
+        assets.add(asset);
+      }
+      return Array.from(assets);
+    }
+    return [];
+  }, [auction]);
+
+  const assetBalances = new Map<string, bigint>();
+  assetKeys.forEach((asset) =>
+    assetBalances.set(
+      asset,
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      useTokenBalance(asset, pool.reserves.get(asset)?.tokenMetadata.asset, horizonAccount).data ??
+        BigInt(0)
+    )
+  );
```

Comment:
thoughts on adding a hook for `useTokenBalances` that intakes an array of `Assets` and returns a set of `string, bigint`? 

Ignoring the rule of hooks can lead to weird behavior in rendering.

Reply:
we probably have two reasonable options:

1. make use of useQueries to write a new hook, `useTokenBalances`, so we can retain the caching of `useTokenBalance`

2. Make a new component called "BalanceChange", which takes in the assetId and balance delta, then just call the individual hook in that component.